tracer/test/Datadog.Trace.Tests/SpanContextPropagatorTests.cs

```diff
@@ -3,34 +3,322 @@
+
+            VerifySetCalls(headers);
+        }
+
+        [Fact]
+        public void Extract_IHeadersCollection()
+        {
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            var headers = SetupMockHeadersCollection();
+            object result = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_CarrierAndDelegate()
+        {
+            // using IHeadersCollection for convenience, but carrier could be any type
+            var headers = SetupMockHeadersCollection();
+
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            object result = SpanContextPropagator.Instance.Extract(headers.Object, (carrier, name) => carrier.GetValues(name));
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_ReadOnlyDictionary()
+        {
+            // using IHeadersCollection for convenience, but carrier could be any type
+            var headers = SetupMockReadOnlyDictionary();
+
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            object result = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_EmptyHeadersReturnsNull()
+        {
+            var headers = new Mock<IHeadersCollection>();
+
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            object resultContext = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            resultContext.Should().BeNull();
+        }
```

Comment:
Is it worth adding a test that _all_ we need for a non-null `SpanContext` is a `TraceId`? i.e. if everything else is missing, we still get distributed tracing?

Reply:
Added `Extract_TraceIdOnly()` in 142fc0f249ef6dbeb4db7f0aa7e44bffefa70e88.